### PR TITLE
Auro-counter #54

### PR DESCRIPTION
**Pull request by rmenner**
# Alaska Airlines Pull Request

## Before Submitting this pull request:
- Link all tickets in this repository related to this PR in the `Development` section
_note: all pull requests require at least one linked ticket_
- If this PR is `Ready For Review`, all ticket's linked under `Development` must have their status changed to `Ready For Review` as well

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I have performed a self-review of my own update.**

## Summary by Sourcery

Add the auro-counter component to the project, complete with documentation and examples for usage and integration.

New Features:
- Introduce the auro-counter component, a UI element for incrementing or decrementing a single digit value, commonly used for adding passengers to a flight.

Documentation:
- Add comprehensive documentation for the auro-counter component, including API details, usage examples, and design notes.

### CODE REVIEW

File: components/dropdown/demo/api.min.js [hunk 1]
```diff
@@ -56,7 +56,7 @@ const t$1=globalThis,e$3=t$1.ShadowRoot&&(void 0===t$1.ShadyCSS||t$1.ShadyCSS.na
 
 /* eslint-disable line-comment-position, no-inline-comments, no-confusing-arrow, no-nested-ternary, implicit-arrow-linebreak */
 
-class AuroLibraryRuntimeUtils {
+let AuroLibraryRuntimeUtils$1 = class AuroLibraryRuntimeUtils {
```
Comment (chrisfalaska) [inline components/dropdown/demo/api.min.js:59]: Was this auto-generated by a process?

Have we used the `$1` convention elsewhere?

File: components/dropdown/demo/api.min.js [hunk 2]
```diff
@@ -2439,7 +2509,7 @@ class AuroDropdown extends r {
     /**
      * @private
      */
-    this.runtimeUtils = new AuroLibraryRuntimeUtils();
+    this.runtimeUtils = new AuroLibraryRuntimeUtils$1();
```
Comment (chrisfalaska) [inline components/dropdown/demo/api.min.js:2512]: Same here re: `$1`. Just want to be sure it's a convention we currently use, and other developers are away of it. It appears throughout the PR.

File: components/dropdown/demo/api.min.js (hunk 1 above)
Comment (rmenner) [reply components/dropdown/demo/api.min.js:59]: Yes this is a generated file. Planning to get these removed from git once I have my new build system in place.

File: components/dropdown/demo/api.min.js (hunk 2 above)
Comment (rmenner) [reply components/dropdown/demo/api.min.js:2512]: This is also a generated file.

**Comment by rmenner** [comment]
Too many rebase issues. Created a new branch.

**Pull request closed without merging**